content/chapters/compute/lab/content/processes.md

```diff
@@ -114,17 +114,47 @@ From the outside, it's as if you were running these commands from the terminal.
 Head over to `support/sleepy/sleepy_creator.py`.
 Use `subprocess.Popen()` to spawn 10 `sleep 1000` processes.
 
-1. Now use the same `pstree -pac` command and look for `sleepy_creator.py`.
-It is a `python3` process, as this is the interpreter that runs the script, but we call it the `sleepy_creator.py` process for simplicity.
-If you found it, you did something wrong.
-It should be missing.
-Now use `pstree -pac` and look for the `sleep` processes you have just created.
+1. Start the script, and look for the parent process using these commands:
 
-[Quiz](../quiz/parent-of-sleep-processes.md)
+   ```console
+   student@os:~$ python3 sleepy_creator.py
+   student@os:~$ ps -e -H -o pid,ppid,cmd | (head -1; grep "python3 sleepy_creator.py")
+   ```
+
+   It is a `python3` process, as this is the interpreter that runs the script, but we call it the `sleepy_creator.py` process for simplicity.
+   You shouldn't be able to find it, as the parent process (`sleepy_creator.py`) dies before its child processes (the 10 `sleep 1000` subprocesses) finish their execution.
+   The parent process of the newly created child processes is an `init`-like process: either `systemd`/`init` or another system process that adopts orphan processes.
+   Now use 
+
+   ```console
+   student@os:~$ ps -e -H -o pid,ppid,cmd | (head -1; grep sleep)
+   ``` 
+   and look for the `sleep` processes you have just created.
+
+   [Quiz](../quiz/parent-of-sleep-processes.md)
 
 1. Change the code in `sleepy_creator.py` so that the `sleep 1000` processes are the children of `sleepy_creator.py`.
-Kill the previously created `sleep` processes using `killall sleep`.
-Verify that `sleepy_creator.py` remains the parent of the `sleep`s it creates using `pstree -pac`.
+   This means that the parent / creator process must **not** exit until its children have finished their execution.
+   In other words, the parent / creator process must **wait** for the termination of its children.
+   Check out [`Popen.wait()`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait) and add the code that makes the parent / creator process wait for its children.
+   Kill the previously created `sleep` processes using `killall sleep`.
+   Verify that `sleepy_creator.py` remains the parent of the `sleep`s it creates using
```

Comment:
First tell them to start the program. Then show the command. Then tell them to inspect the process, then run the `ps` command.